appengine/flexible/tasks/appengine_queues/app_engine_queue_snippets.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import base64
+import datetime
+
+from googleapiclient import discovery
+
+
+def format_rfc3339(datetime_instance):
+    """Format a datetime per RFC 3339."""
+    return datetime_instance.isoformat("T") + "Z"
+
+
+def get_seconds_from_now_rfc3339(seconds):
+    """Return seconds from the current time as a RFC 3339 string."""
+    d = datetime.datetime.utcnow() + datetime.timedelta(seconds=seconds)
+    return format_rfc3339(d)
+
+
+def list_queues(api_key, project_id, location_id):
+    """List the queues in the location."""
+    client = get_client(api_key)
+    parent = 'projects/{}/locations/{}'.format(project_id, location_id)
+    queues = []
+    next_page_token = None
+
+    while True:
+        response = client.projects().locations().queues().list(
+            parent=parent, pageToken=next_page_token).execute()
+        queues += response['queues']
+        if next_page_token is None:
+            break
+
+    print('Listing queues for location {}'.format(location_id))
+
+    for queue in response['queues']:
+        print queue['name']
+    return response
+
+
+def create_task(api_key, queue_name, payload=None, in_seconds=None):
+    """Create a task for a given queue with an arbitrary payload."""
+    client = get_client(api_key)
+
+    url = '/set_payload'
+    task = {
+        'task': {
+            'app_engine_task_target': {
+                'http_method': 'POST',
+                'relative_url': url
+            }
+        }
+    }
+
+    if payload is not None:
+        task['task']['app_engine_task_target']['payload'] = base64.b64encode(
+            payload)
+
+    if in_seconds is not None:
+        scheduled_time = get_seconds_from_now_rfc3339(int(in_seconds))
```

Comment:
why the `int` cast here? fractional seconds are valid.

Reply:
Done